### DISCUSSION TITLE
请教mlr3机器学习回归问题

### DISCUSSION THREAD

**Discussion by cinnamonfish** [ShixiangWang/self-study#112] [Q&A] [answered]
您好，我想请教一个R代码运行报错：
我尝试用mlr3verse去做regr.xgboost模型的训练，因变量是连续值，自变量有连续变量和分类变量（因子变量）。但我遇到了如下报错：“ Error: <TaskRegr:id> has the following unsupported feature types: factor >”。想请问一下报错是什么意思，是说自变量不能包含因子变量吗？

**Comment by ShixiangWang** [comment, marked as answer]
看起来是这样的，但 R 对因子的支持一般都相当好的，不应该出现这种问题。不过默认的 xgboost 包确实是不支持因子的，它需要一个数值矩阵，所以可能你需要做下 one-hot 编码。你也可以通过去除自变量的分类变量看是不是不会报这个错误来进行确定。

**Reply by cinnamonfish** [reply]
非常感谢，我做了one hot encoding就不报错了。

**Reply by ShixiangWang** [reply]
好的，谢谢回复哈
